vulcanai2/tests/models/test_device.py

```diff
@@ -139,10 +140,31 @@ def test_master_net_device_set_to_cuda(self, multi_net):
         assert multi_net.input_networks['dense_net'].input_networks['conv2D_net']\
             .device == torch.device(type='cuda', index=0)                                    
 
-    # @pytest.mark.xfail
-    # def test_fail_mixed_devices(self, multi_net):
-    # TODO: write script with mixed devices so that the 
-    # network fails when fitting
+    def test_fail_mixed_devices(self, multi_net):
+        master_device_setter(multi_net, 'cuda:0')
+        multi_net.input_networks['dense_net'].input_networks['conv2D_net'].device = "cpu"
+
+        data = MultiDataset([
+            (TensorDataset(torch.ones([10, *multi_net.input_networks['conv3D_net'].in_dim])), True, False),

```

Comment:
@priyatharsan any way to shorten the call to input networks? perhaps having a ref link to them directly?

Reply:
yeah that could work but that will eventually increases the parameters to the function, is that okay?

Reply:
that's fine. the functions aren't meant to be used by anyone but the testing environment. Plus you can also do assertions that the model references are the same. e.g. `assert conv_2D_net == multi_input.input_networks['conv_2D_net']`